docs/getting_started/basics/execute.md

```diff
@@ -1,31 +1,35 @@
+## ![Execute All Button](../../images/toolbar/execute_all_default.svg){:width="25" height="25"} Execute All
 
-![Execute All](../images/execute_all.gif)
+The "Execute All" action will compute all the nodes present in the workflow. 
 
-## Stop Execution
+![Execute All](../../images/execute_all.gif)
 
-Shift uses dirty propagation to determine which nodes need to be recomputed. Whenever a plug value on a node is modified, that node is set to be recomputed in the next execution. You can force Shift to recompute all nodes by pressing the “Stop Execution” button (![Stop Execution](../images/toolbar/button_executeStop.png){:width="30" height="30"}).
+## ![Execute Selected Button](../../images/toolbar/execute_selected_default.svg){:width="25" height="25"} Execute Selected
 
-## Execute Step-by-Step
+The "Execute Selected" action will compute all the selected nodes in the workflow. To do so, Shift will identify the workflow sub-graph required to evaluate the desired nodes and execute it.
 
-It is also possible to execute a workflow step-by-step. Simply press the “Execute Next Operator” button (![Execute Next Button](../images/toolbar/button_executeNext.png){:width="30" height="30"}) to see the workflow being executed one node at a time.
+![Execute Selected](../../images/execute_selected.gif)
 
-![Execute Step by Step](../images/execute_step_by_step.gif)
+## ![Execute Next Button](../../images/toolbar/execute_next_default.svg){:width="25" height="25"} Execute Next
 
-## Execute Selected
+The "Execute Next" action will compute the next dirty node present in the workflow. This allows to execute one node at a time and becomes useful to closely follow and inspect the processing of the workflow.
 
-The execution can also be forced on selected nodes. By selecting the desired nodes and pressing the “Execute Selected” button (![Execute Selected Button](../images/toolbar/button_executeSelected.png){:width="30" height="30"}), Shift will identify the workflow sub-graph required to evaluate the desired nodes and execute it.
+![Execute Next](../../images/execute_step_by_step.gif)
 
-## Live Execution
+## ![Stop Execution Button](../../images/toolbar/clear_all_default.svg){:width="25" height="25"} Stop Execution
 
-Finally, it is also possible to trigger the execution of a workflow whenever a change to any node is done by using the live execution option. Press the “Toggle Live Execution” button (![Live Execute](../images/toolbar/button_executeLiveOff.png){:width="30" height="30"}) and Shift will execute the workflow automatically every time a change is being made.
+The "Stop Execution" action will set all the nodes in the workflow to dirty. All **connected** inputs and output values will be cleared to their default values. This forces Shift to recompute all nodes the next time the workflow has to be executed. 
```

Comment:
At the moment, we are calling "Stop Execution" to the action of clearing the workflow. The button does not stop the execution perse, and the icon does not indicate that that's the behavior of the button anyways.  
Should we change in Shift the name of the button so it doesn't cause confusion?

Reply:
I think some time ago we had a similar conversation, and we decided that because when threading would be in place the behavior will be the one to stop, the button should keep the same name.

However, my point is the same as at that discussion. The button is cleaning the execution stack, and would have much sense for the user to be a "Reset" or a "Clean" button.

Reply:
Right now it is more `reset execution`. But I would keep it as stop execution because once we have implemented threading it will be stop execution.

Reply:
Issue has been created
https://github.com/Inbibo/shift/issues/1102